tests/e2e_test.go

```diff
@@ -0,0 +1,565 @@
+			GasFeeCap: gasFeeCapA,
+			GasTipCap: gasTipCapA,
+			Value:     value,
+		})
+		txSignerA := types.LatestSignerForChainID(chainAIDInt)
+		triggerTxA, err := types.SignTx(txA, txSignerA, fundedKey)
+		Expect(err).Should(BeNil())
+		err = chainARPCClient.SendTransaction(ctx, triggerTxA)
+		Expect(err).Should(BeNil())
+		time.Sleep(5 * time.Second)
+		receipt, err := chainARPCClient.TransactionReceipt(ctx, triggerTxA.Hash())
+		Expect(err).Should(BeNil())
+		Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+	}
+	{
+		value := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(10)) // 10eth
+		txB := types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainBIDInt,
+			Nonce:     nonceB,
+			To:        &teleporterDeployerAddress,
+			Gas:       defaultTeleporterMessageGas,
+			GasFeeCap: gasFeeCapB,
+			GasTipCap: gasTipCapB,
+			Value:     value,
+		})
+		txSignerB := types.LatestSignerForChainID(chainBIDInt)
+		triggerTxB, err := types.SignTx(txB, txSignerB, fundedKey)
+		Expect(err).Should(BeNil())
+		err = chainBRPCClient.SendTransaction(ctx, triggerTxB)
+		Expect(err).Should(BeNil())
+		time.Sleep(5 * time.Second)
+		receipt, err := chainBRPCClient.TransactionReceipt(ctx, triggerTxB.Hash())
+		Expect(err).Should(BeNil())
+		Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+	}
+	// Deploy Teleporter on the two subnets
+	{
+		rpcClient, err := rpc.DialContext(ctx, chainARPCURI)
+		Expect(err).Should(BeNil())
+		err = rpcClient.CallContext(ctx, nil, "eth_sendRawTransaction", hexutil.Encode(teleporterDeployerTransaction))
+		Expect(err).Should(BeNil())
+		time.Sleep(5 * time.Second)
+		teleporterCode, err := chainARPCClient.CodeAt(ctx, teleporterContractAddress, nil)
+		Expect(err).Should(BeNil())
+		Expect(len(teleporterCode)).Should(BeNumerically(">", 2)) // 0x is an EOA, contract returns the bytecode
+	}
+	{
+		rpcClient, err := rpc.DialContext(ctx, chainBRPCURI)
+		Expect(err).Should(BeNil())
+		err = rpcClient.CallContext(ctx, nil, "eth_sendRawTransaction", hexutil.Encode(teleporterDeployerTransaction))
+		Expect(err).Should(BeNil())
+		time.Sleep(5 * time.Second)
+		teleporterCode, err := chainBRPCClient.CodeAt(ctx, teleporterContractAddress, nil)
+		Expect(err).Should(BeNil())
+		Expect(len(teleporterCode)).Should(BeNumerically(">", 2)) // 0x is an EOA, contract returns the bytecode
+	}
+	log.Info("Finished deploying Teleporter contracts")
+
+	log.Info("Set up ginkgo before suite")
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	log.Info("Running ginkgo after suite")
+	Expect(manager).ShouldNot(BeNil())
+	Expect(manager.TeardownNetwork()).Should(BeNil())
+	Expect(os.Remove(warpChainConfigPath)).Should(BeNil())
+})
+
+// Ginkgo describe node that acts as a container for the teleporter e2e tests. This test suite
+// will run through the following steps in order:
+// 1. Send a transaction to the Teleporter contract on Subnet A
+// 2. Aggregate signatures and send the Warp message to Subnet B
```

Comment:
can you add that it verifies the received warp message values

Reply:
I actually think we should remove Warp message verification, since the interface we're testing end-to-end is Teleporter, not Warp. Explicit Warp message verification is best left up to subnet-evm. I've removed this now.